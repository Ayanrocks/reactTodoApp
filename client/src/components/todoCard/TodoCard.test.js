import React from "react";
import TodoCard from "./index";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

it("TodoCard renders correctly", () => {
  const mockTodo = {
    id: 1,
    name: "Hello Todo",
    status: "pending",
    bucket: "default"
  };
  function handleDelete() {
    console.log("delete clicked");
  }
  const tree = renderer
    .create(
      <BrowserRouter>
        <TodoCard todo={mockTodo} handleDelete={handleDelete} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
