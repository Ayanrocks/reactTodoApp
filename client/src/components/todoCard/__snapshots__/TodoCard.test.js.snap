// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoCard renders correctly 1`] = `
<div
  className="card shadow card-shadow my-5 mx-3"
  style={
    Object {
      "width": "15rem",
    }
  }
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Hello Todo
    </h5>
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-around",
        }
      }
    >
      <a
        className="btn btn-success"
        href="/todo/1"
        onClick={[Function]}
      >
        <span
          className="fa fa-eye"
        />
      </a>
      <button
        className="btn btn-danger"
        onClick={[Function]}
      >
        <span
          className="fa fa-trash"
        />
      </button>
    </div>
  </div>
</div>
`;
